cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -259,6 +267,34 @@ func (impl *OSBuildJobImpl) getContainerClient(destination string, targetOptions
 	return client, nil
 }
 
+func (impl *OSBuildJobImpl) getPulpClient(targetOptions *target.PulpOSTreeTargetOptions) (*pulp.Client, error) {
+	var creds *pulp.Credentials
+	if targetOptions.Username != "" && targetOptions.Password != "" {
+		creds = &pulp.Credentials{
+			Username: targetOptions.Username,
+			Password: targetOptions.Password,
+		}
+	}
+	address := targetOptions.ServerAddress
+	if address == "" {
+		address = impl.PulpConfig.ServerAddress
+	}
+	if address == "" {
+		return nil, fmt.Errorf("pulp server address not set")
+	}
+
+	if creds != nil {
+		return pulp.NewClient(address, creds), nil
+	}
+
+	// read from worker configuration
+	if impl.PulpConfig.CredsFilePath == "" {
+		return nil, fmt.Errorf("pulp credentials not set")
+	}
+
+	return pulp.NewClientFromFile(impl.PulpConfig.ServerAddress, impl.PulpConfig.CredsFilePath)
```

Comment:
Shouldn't you use `address` instead of `impl.PulpConfig.ServerAddress`? I think that I get your intention to not use user-provided server address with credentials from the configuration. However it can be used the other way around - use server address from the configuration with user-provided credentials.

Is this intentional?

Reply:
I can't remember if I meant to do it this way for the reason you mentioned, but I think you're right and I should change it.